Fix docker image always changed even if image has not been updated.

cuckoohello opened the pull request:
While using force mode in docker module,  always return Changed=True even if the image has not been updated.

This pull request fixes this issue, so we can always use force in docker image and recreate docker container only if the upstream image is updated.

Commit:
Fix docker image always changed even if image has not been updated while using force mode

gregdek commented:
Thanks @cuckoohello for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

chouseknecht commented:
@cuckoohello 

Can you provide an example playbook that produces the bug. I'd like to reproduce the error and test this PR before merging it. An example will make it easier.

Thank you for doing the work on this and bringing it to our attention. Nice job!

cuckoohello commented:
```
---

- name: Test docker image always changed even if image has not been updated bug while using force mode.
  hosts: servers
  tasks:
    - name: Download image alpine first time
      docker_image:
        name: alpine
        force: yes
        state: present

    - name: Download image alpine second time with same config
      docker_image:
        name: alpine
        force: yes
        state: present
```

Using the playbook above, they are two tasks using the same config except the name.

Without my patch, the outpout looks like below:

```
PLAY [Test docker image always changed even if image has not been updated bug while using force mode.] ***

TASK [setup] *******************************************************************
ok: [btsync]

TASK [Download image alpine first time] ****************************************
changed: [btsync]

TASK [Download image alpine second time with same config] **********************
changed: [btsync]

PLAY RECAP *********************************************************************
btsync                     : ok=3    changed=1    unreachable=0    failed=0 
```

The task "Download image alpine second time with same config" should return ok because the upstream docker image alpine is not updated.

With the patch, it will looks like：

```
PLAY [Test docker image always changed even if image has not been updated bug while using force mode.] ***

TASK [setup] *******************************************************************
ok: [btsync]

TASK [Download image alpine first time] ****************************************
changed: [btsync]

TASK [Download image alpine second time with same config] **********************
ok: [btsync]

PLAY RECAP *********************************************************************
btsync                     : ok=3    changed=1    unreachable=0    failed=0   
```

Only the first pull should result in changed(alpine image is not existed in the server before test),  task "Download image alpine second time with same config" will detect the image id is as same as previous, so it should return ok.

cuckoohello commented:
@chouseknecht 
Did you have time to test this pull request?
